rt/src/test/java/io/fabric8/maven/rt/SpringbootCrudBoosterIT.java

```diff
@@ -0,0 +1,140 @@
+        Repository testRepository = setupSampleTestRepository(SPRING_BOOT_CRUD_BOOSTER_GIT, RELATIVE_POM_PATH);
+        deployDatabaseUsingCLI();
+
+        deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
+        waitAfterDeployment(false);
+        assertApplication();
+    }
+
+    @Test
+    public void redeploy_springboot_app() throws Exception {
+        Repository testRepository = setupSampleTestRepository(SPRING_BOOT_CRUD_BOOSTER_GIT, RELATIVE_POM_PATH);
+        deployDatabaseUsingCLI();
+
+        // Make some changes in ConfigMap and rollout
+        updateSourceCode(testRepository, RELATIVE_POM_PATH);
+        addRedeploymentAnnotations(testRepository, RELATIVE_POM_PATH, ANNOTATION_KEY, ANNOTATION_VALUE, fmpConfigurationFile);
+
+        deploy(testRepository, EMBEDDED_MAVEN_FABRIC8_BUILD_GOAL, EMBEDDED_MAVEN_FABRIC8_BUILD_PROFILE);
+        waitAfterDeployment(true);
+        assertApplication();
+    }
+
+    private void deploy(Repository testRepository, String buildGoal, String buildProfile) throws Exception {
+        runEmbeddedMavenBuild(testRepository, buildGoal, buildProfile);
+    }
+
+    private void assertApplication() throws Exception {
+        assertThat(openShiftClient).deployment(testsuiteRepositoryArtifactId);
+        assertThat(openShiftClient).service(testsuiteRepositoryArtifactId);
+
+        RouteAssert.assertRoute(openShiftClient, testsuiteRepositoryArtifactId);
+        executeCRUDAssertions(getApplicationRouteWithName(testsuiteRepositoryArtifactId));
+    }
+
+    private void waitAfterDeployment(boolean bIsRedeployed) throws Exception {
+        // Waiting for application pod to start.
+        if (bIsRedeployed)
+            waitTillApplicationPodStarts(ANNOTATION_KEY, ANNOTATION_VALUE);
+        else
+            waitTillApplicationPodStarts();
+        // Wait for Services, Route, ConfigMaps to refresh according to the deployment.
+        TimeUnit.SECONDS.sleep(20);
+    }
+
+    private void executeCRUDAssertions(Route applicationRoute) throws Exception {
+        String hostUrl = "http://" + applicationRoute.getSpec().getHost() + TEST_ENDPOINT;
+        JSONObject jsonRequest = new JSONObject();
+        String testFruitName = "Pineapple";
+
+        // (C) Create
+        jsonRequest.put("name", testFruitName);
+        Response createResponse = makeHttpRequest(HttpRequestType.POST, hostUrl, jsonRequest.toString());
+        assert createResponse.code() == HttpStatus.SC_CREATED;
+        Integer fruitId = new JSONObject(createResponse.body().string()).getInt("id");
+        hostUrl += ("/" + fruitId.toString());
+
+        // (R) Read
+        Response readResponse = makeHttpRequest(HttpRequestType.GET, hostUrl, null);
+        String fruitName = new JSONObject(readResponse.body().string()).getString("name");
+        assert readResponse.code() == HttpStatus.SC_OK;
+        assert fruitName.equals(testFruitName);
+
+        // (U) Update
```

Comment:
Read is sufficient to test. We should limit our test scope. Let's get rid of create, update, delete assertions.